테스트코드 리팩토링 및 pipe 테스트코드 구현

KimJye opened the pull request:
## 작업 내용
video-list-query-pipe, user-list-query-pipe, tag-list-query-pipe에 대한 테스트코드를 작성했습니다.
video.service, user.service, tag.service가 팀원들의 기능 구현들로인해 테스트 코드의 수정이 필요하여 수정했습니다.

## 실행 방법
npm test

KimJye committed:
[server] | feat(user-list-pipe-test): user-list-query-pipe 테스트코드 구현

user-list-query-pipe에대한 테스트코드를 구현했습니다.
더불어 user.service에 videoRepository가 새롭게 주입되어 관련
테스트코드도 수정하였습니다.

KimJye committed:
[server] | feat(video-list-query-pipe): video-list-query-pipe 테스트코드

pipe에서 각 조건에따른 테스트를 진행하였습니다.
videoListQueryDto에서 sort는 undefind 경우가 있으므로 ?를
추가하였습니다.
videoController에 try catch를 추가하였습니다.

KimJye committed:
[server] | refactor(video): video injection 수정에따른 테스트코드 수정

video service에 comment와 upload 관련 코드가 추가되어 테스트코드도
수정하였습니다.
video.controller에서 try catch로 감싸주었습니다.

riverandeye commented on packages/server/src/user/pipe/user-list-query-pipe.spec.ts:
```diff
@@ -0,0 +1,35 @@
+
+  describe('success calls', () => {
+    it('should let the UserListQueryDto go on through', () => {
+      const userListQueryDto = new UserListQueryDto(1, '권');
```
UserListQueryDto(1, '권')에서 1을 string 으로 입력해야 할 것 같습니다 ^^7

KimJye committed:
[server] | refactor(pipe-test): page 테스트 값을 string으로 변경

query string으로 넘어오는 값들은 string이고 pipe 결과로 int형을
반환해야하기 때문에 string으로 테스트 값을 변경하였습니다.

riverandeye commented on packages/server/src/video/pipe/video-list-query-pipe.ts:
```diff
@@ -11,7 +11,7 @@ export class VideoListQueryPipe implements PipeTransform {
     if (!this.validateGetVideosPipeDto({ page, sort, period, keyword })) {
-      throw new BadRequestException();
+      throw new BadRequestException('should valid query string');
     }
```
리턴하는 에러 메세지로 Query String is not valid 는 어떨까요?? 
should valid query string이 말이 쪼끔 어색한거같아요

KimJye replied:
넵~

riverandeye reviewed (approved):
덕분에 많이 배웁니다. 감사해요 지혜님~

KimJye committed:
[server] | refactor(videolistquerypipespec): 에러 메시지 변경

hyokualexkwon commented on packages/server/src/tag/tag.controller.ts:
```diff
@@ -14,8 +19,12 @@ export class TagController {
+      return new TagListResponseDto(tags, count);
+    } catch (err) {
+      throw new InternalServerErrorException(err.message);
+    }
```
try catch를 반드시 써야 할까요?

KimJye replied:
혹시 모를 서버 에러에 대처하기 위함입니당

KimJye replied:
BadRequestException으로 변경했습니당

hyokualexkwon commented on packages/server/src/user/user.controller.ts:
```diff
@@ -40,9 +41,12 @@ export class UserController {
+      return new UserListResponseDto(users, count);
+    } catch (err) {
+      throw new InternalServerErrorException(err.message);
+    }
```
try catch 반드시 작성해야 하나요?

KimJye replied:
혹시 모를 서버 에러에 대처하기 위함입니당

hyokualexkwon commented on packages/server/src/video/video.controller.ts:
```diff
@@ -49,11 +50,15 @@ export class VideoController {
+      return new VideoListResponseDto(videos, count);
+    } catch (err) {
+      throw new InternalServerErrorException(err.message);
+    }
```
try catch 반드시 작성해야 하나요?

KimJye replied:
혹시 모를 서버 에러에 대처하기 위함입니당

KimJye committed:
[server] | refactor(controller): BadRequest Error로 변경

KimJye committed:
Merge branch 'develop' of https://github.com/connect-foundation/2019-10 into refactor/search_api_20191217

hyokualexkwon commented on packages/server/src/tag/pipe/tag-list-query-pipe.spec.ts:
```diff
@@ -0,0 +1,35 @@
+  });
+
+  describe('success calls', () => {
+    it('should let the TagListQueryDto go on through', () => {
```
`it should let TagListQueryDTO to go through`라고 작성하면 더 좋을것 같습니다 ㅎㅎ

KimJye replied:
키야 역시..! 배우고 갑니다

hyokualexkwon commented on packages/server/src/tag/pipe/tag-list-query-pipe.spec.ts:
```diff
@@ -0,0 +1,35 @@
+  });
+
+  describe('unsuccess calls', () => {
+    it('should throw an error if given a non-numberic page', () => {
```
`it should throw an error if non-numeric page is given`이라고 작성하면 더 좋을것 같습니다 ㅎㅎ

hyokualexkwon commented on packages/server/src/user/pipe/user-list-query-pipe.spec.ts:
```diff
@@ -0,0 +1,35 @@
+  });
+
+  describe('success calls', () => {
+    it('should let the UserListQueryDto go on through', () => {
```
ㅎㅎ

hyokualexkwon commented on packages/server/src/user/pipe/user-list-query-pipe.spec.ts:
```diff
@@ -0,0 +1,35 @@
+  });
+
+  describe('unsuccess calls', () => {
+    it('should throw an error if given a non-numberic page', () => {
```
ㅎㅎ

hyokualexkwon commented on packages/server/src/video/pipe/video-list-query-pipe.spec.ts:
```diff
@@ -0,0 +1,114 @@
+  });
+
+  describe('success calls', () => {
+    it('should let the VideoListQueryDto go on through with keyword', () => {
```
`it should properly contain keyword when it is given on VideoListQueryDto` 는 어떨까요?ㅎㅎ

KimJye replied:
ㅎㅎ 감사합니다

KimJye committed:
[server] | refactor(test): 메시지 변경
